doc/string.rdoc

```diff
@@ -47,4 +47,81 @@ class String
   def initialize(str = '', encoding: nil, capacity: nil)
     Primitive.rb_str_init(str, encoding, capacity)
   end
+
+  #  call-seq:
+  #    split(pattern = $FIELD_SEPARATOR, limit = nil) -> array
+  #    split(pattern = $FIELD_SEPARATOR, limit = nil) {|sub| block } -> object
```

Comment:
I don't think `pattern` is `$FIELD_SEPARATOR` by default:

```ruby
irb(main):004:0> "foo bar".split
=> ["foo", "bar"]
irb(main):005:0> $FIELD_SEPARATOR = "o"
=> "o"
irb(main):006:0> "foo bar".split
=> ["foo", "bar"]
```

I think `$;` is not the same as `$FIELD_SEPARATOR`:

```ruby
irb(main):008:0> $; = "o"
=> "o"
irb(main):009:0> "foo bar".split
=> ["f", "", " bar"]
```

I think the English library needs to be loaded for this to work:

```ruby
irb(main):010:0> require "English"
=> true
irb(main):011:0> $FIELD_SEPARATOR = "a"
=> "a"
irb(main):012:0> "foo bar".split
=> ["foo b", "r"]
```

Reply:
I have changed $FIELD_SEPARATOR to $; throughout.  Also have changed argument name from +pattern+ to +separator+, which actually describes the function (instead of just the form).